parsing/dialects/dialect1/parserImpls.ftl

```diff
@@ -1792,6 +1792,55 @@ SqlNode SqlSelectTopN(SqlParserPos pos) :
     }
 }
 
+SqlNode InlineCaseSpecific() :
+{
+    final SqlNode value;
+    boolean not = false;
+}
+{
+    (
+        value = StringLiteral()
+    |
+        LOOKAHEAD( CompoundIdentifier() <LPAREN> [ <NOT> ] (<CASESPECIFIC>|<CS>) <RPAREN> )
```

Comment:
Can we put this parsing logic in a helper function (e.g. called "CaseSpecific") that you can call both from the LOOKAHEAD and below?

Reply:
Good idea, will do.